vault/version_store.go

```diff
@@ -149,6 +151,40 @@ func (c *Core) loadVersionHistory(ctx context.Context) error {
 	return nil
 }
 
+// isMajorVersionFirstMount compares the current version of Vault with the last
+// version that was successfully mounted and unsealed to see if it is a
+// milestone or major upgrade. This is useful in determining shutdown behavior
+// for deprecated builtins.
+func (c *Core) isMajorVersionFirstMount(ctx context.Context) (bool, error) {
+	newestVersion, _, err := c.FindNewestVersionTimestamp(ctx)
```

Comment:
Shouldn't the newestVersion be the same as curr always?

Reply:
This is comparing the newest _recorded_ version from the version history. `curr` should not be stored until we successfully unseal.

Reply:
Got it, ty.